cmd/mayactl/app/command/volumes_list.go

```diff
@@ -52,30 +74,56 @@ func NewCmdVolumesList() *cobra.Command {
 	return cmd
 }
 
-//RunVolumesList fetchs the volumes from maya-apiserver
+//RunVolumesList fetches the volumes from maya-apiserver
 func (c *CmdVolumeOptions) RunVolumesList(cmd *cobra.Command) error {
-	//fmt.Println("Executing volume list...")
 
-	var cvols v1alpha1.CASVolumeList
-	err := mapiserver.ListVolumes(&cvols)
+	// Calling to m-api service to fetch volume list.
+	cvols, err := mapiserver.ListVolumes()
 	if err != nil {
 		return fmt.Errorf("Volume list error: %s", err)
 	}
 
-	out := make([]string, len(cvols.Items)+1)
-	out[0] = "Name|Status|Type"
-	for i, items := range cvols.Items {
-		if len(items.Status.Reason) == 0 {
-			items.Status.Reason = volumeStatusOK
+	// Create a slice of VolumeList struct that can be binded to template.
+	volumes := []VolumeList{}
+
+	// Filling the slice with required fields.
+	for _, vol := range cvols.Items {
+		volume := VolumeInfo{
+			Volume: vol,
 		}
-		out[i+1] = fmt.Sprintf("%s|%s|%s",
-			items.ObjectMeta.Name,
-			items.Status.Reason, items.Spec.CasType)
+		volumes = append(volumes, VolumeList{
+			Namespace:  volume.GetVolumeNamespace(),
+			Name:       volume.GetVolumeName(),
+			Status:     volume.GetVolumeStatus(),
+			VolumeType: strings.Title(volume.GetCASType()),
+		})
+	}
+
+	// Check for volumes length.
+	if len(volumes) == 0 {
+		fmt.Println("No volumes found")
+		return nil
 	}
-	if len(out) == 1 {
-		fmt.Println("No Volumes are running")
+
+	// Creating template instance and parsing structure into it.
+	volumeListTemplate, err := template.New("VolumeList").Parse(listTemplate)
+	if err != nil {
+		fmt.Println("Error displaying output, found error:", err)
 		return nil
 	}
-	fmt.Println(util.FormatList(out))
+
+	// Creating tabwriter instance and executing it with template.
+	w := tabwriter.NewWriter(os.Stdout, v1.MinWidth, v1.MaxWidth, v1.Padding, ' ', 0)
```

Comment:
Why is tabwriter being used?
Have we just tried with go templating minus tabwriter ?

Reply:
@AmitKumarDas i guess, tabwriter is needed because sometimes if the length of values vary the output gets destorted.

Reply:
ok.. thanks @utkarshmani1997 I guess this tabwriter can be built as a template function & used in the template itself.
The whole point separating the rendering logic from business logic.

Reply:
sounds good.